modules/csvparser/filterx-func-parse-csv.c

```diff
@@ -114,21 +134,17 @@ _eval(FilterXExpr *s)
                             FILTERX_FUNC_PARSE_CSV_ARG_NAME_STRING_DELIMITERS))
     goto exit;
 
-  if (string_delimiters)
-    csv_scanner_options_set_string_delimiters(&self->options, string_delimiters);
-
   if (!_parse_list_argument(self, self->columns, &cols, FILTERX_FUNC_PARSE_CSV_ARG_NAME_COLUMNS))
     goto exit;
 
   if (cols)
-    {
-      csv_scanner_options_set_expected_columns(&self->options, g_list_length(cols));
-      result = filterx_json_object_new_empty();
-    }
+    result = filterx_json_object_new_empty();
   else
     result = filterx_json_array_new_empty();
 
-  csv_scanner_init(&scanner, &self->options, input);
+  CSVScanner scanner;
+  CSVScannerOptions local_opts = {0};
```

Comment:
if we are not initializing scanner, we might get away with not initializing local_opts either.

Reply:
I need to null-init the scanner, because I call its destructor unconditionally.